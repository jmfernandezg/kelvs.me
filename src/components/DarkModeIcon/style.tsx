import styled from "styled-components";
import { animated } from "react-spring";

export const IconContainer = styled.div`
  display: flex;
  align-items: center;
  justify-self: flex-end;
  justify-content: center;
  min-width: 93.563px;
  ${({ theme }) => theme.tablet`
    display:none;
    min-width: unset;
    flex: 0 1 20%;
    margin-left: 35px;
    `}
`;

export const AnimatedSpan = styled(animated.span)`
  display: inherit;
  svg {
    width: auto;
    height: 1.5em;
    cursor: pointer;
    fill: ${props => props.theme.text.dark};
    .dark-mode & {
      fill: ${props => props.theme.text.light};
    }
  }
`;
