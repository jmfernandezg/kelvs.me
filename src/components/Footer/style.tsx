import styled from "styled-components";

const FooterContainer = styled.footer`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: wrap;
  min-height: 3em;
  font-size: 12px;
  text-align: center;
  color: #999;
  a {
    text-decoration: underline;
    color: #8ccbff;
  }
  ${({ theme }) => theme.thone`font-size: 0.7em;`}
`;

export default FooterContainer;
