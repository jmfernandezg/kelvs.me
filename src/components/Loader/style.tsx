import styled from "styled-components";

export const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: ${props => props.theme.bg.light};
  color: ${props => props.theme.text.dark};
  position: fixed;
  width: 100%;
  height: 100%;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 99;

  .dark-mode & {
    background-color: ${props => props.theme.bg.dark};
    color: ${props => props.theme.text.light};
  }
`;

export const LogoWrapper = styled.div<{ isMounted: boolean }>`
  width: max-content;
  transition: all 0.25s cubic-bezier(0.645, 0.045, 0.355, 1);
  opacity: ${props => (props.isMounted ? 1 : 0)};
  svg {
    width: 100%;
    height: 100%;
    display: block;
    margin: 0 auto;
  }
`;
