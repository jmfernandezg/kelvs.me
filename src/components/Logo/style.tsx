import styled from "styled-components";

const LogoContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  justify-self: flex-start;
  z-index: 20;
  svg {
    width: auto;
    height: 1.5em;
    display: block;
  }
`;

export default LogoContainer;
