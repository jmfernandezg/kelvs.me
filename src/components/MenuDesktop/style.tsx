import styled from "styled-components";

export const MenuDesktopContainer = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  ${({ theme }) => theme.tablet`display:none;`}
`;

export const LinkWrapper = styled.span<{ isActive: boolean }>`
  a {
    text-align: center;
    margin: 0 2px;
    padding: 8px 32px;
    border-radius: 4px;
    background: ${props => (props.isActive ? "rgba(0,122,255, 0.1)" : "none")};
    color: ${props => (props.isActive ? "#007AFF" : "#050505")};
  }
  a:hover {
    color: #007aff;
    background: ${props => props.theme.hover.light};
  }
  .dark-mode & a {
    color: ${props => props.theme.text.light};
    background: ${props => (props.isActive ? props.theme.hover.dark : "none")};
    &:hover {
      background: ${props => props.theme.hover.dark};
    }
  }
`;
