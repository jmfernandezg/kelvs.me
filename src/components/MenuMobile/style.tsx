import styled from "styled-components";

export const MenuMobileContainer = styled.div<{ isActive: boolean }>`
  display: none;
  width: 100%;
  height: 100vh; /* Use vh as a fallback for browsers that do not support Custom Properties */
  height: calc(var(--vh, 1vh) * 100);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: 0 auto;
  overflow: hidden;
  backdrop-filter: blur(8px);
  background-color: ${props => props.theme.bg.light};
  opacity: ${props => (props.isActive ? 0.9 : 0)};
  z-index: ${props => (props.isActive ? 15 : 0)};
  ${({ theme }) => theme.tablet`
    display:flex;
    flex-direction: column;
    align-items: baseline;
    justify-content: space-evenly;
    div:nth-child(2) {
      display: block !important;
    }
  `}
  .dark-mode & {
    background-color: ${props => props.theme.bg.dark};
    span {
      color: ${props => props.theme.bg.light};
    }
  }
  svg {
    opacity: ${props => (props.isActive ? 1 : 0)};
    transition: 700ms;
  }
  @media (prefers-reduced-motion: no-preference) {
    transition: opacity 500ms;
  }
`;

export const LinkWrapper = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  font-size: 28px;
  width: 100%;
  transition: opacity 0.25s ease-in-out;
  span {
    text-align: left;
    display: block;
    color: ${props => props.theme.text.dark};
  }
  a {
    display: inline-block;
    position: relative;
    padding: 15px 0;
    text-decoration: none;
  }
`;

export const HamburgerWrapper = styled.div<{ isActive: boolean }>`
  display: none;
  ${({ theme }) => theme.tablet`display:block;`}
  svg {
    position: relative;
    height: 2em;
    cursor: pointer;
    -webkit-tap-highlight-color: transparent;
    transition: transform 400ms;
    user-select: none;
    z-index: 20;

    .line {
      fill: none;
      transition: stroke-dasharray 400ms, stroke-dashoffset 400ms;
      stroke: currentColor;
      stroke-width: 5.5;
      stroke-linecap: round;
    }
  }

  svg .top {
    stroke-dasharray: 40 172;
  }
  svg .middle {
    stroke-dasharray: 40 111;
  }
  svg .bottom {
    stroke-dasharray: 40 172;
  }

  ${props =>
    props.isActive
      ? `svg .top {
      stroke-dashoffset: -132px;
    }
    svg .middle {
      stroke-dashoffset: -71px;
    }
    svg .bottom {
      stroke-dashoffset: -132px;
    }`
      : ""}
`;
