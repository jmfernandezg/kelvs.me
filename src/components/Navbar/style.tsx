import styled from "styled-components";

export const NavContainer = styled.header`
  display: flex;
  justify-content: space-around;
  position: relative;
  padding: 8px 8px;
  margin: 0;
  top: 0;
  left: 0;
  right: 0;
  font-size: 1em;
  font-weight: 600;
  color: ${props => props.theme.text.dark};
  .dark-mode & {
    color: ${props => props.theme.text.light};
  }
`;

export const MenuWrapper = styled.nav`
  display: flex;
  justify-content: center;
  align-content: center;
  align-items: center;
  width: 100%;
  max-width: 868px;
  ${({ theme }) => theme.tablet`justify-content: space-between;`}
`;
