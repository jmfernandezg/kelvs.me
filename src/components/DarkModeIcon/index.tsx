import useDarkMode from "use-dark-mode";
import { useSpring } from "react-spring";
import { useLoaded } from "~/helpers/index";
import { IconContainer, AnimatedSpan } from "./style";
import SunSVG from "~/components/Icons/Sun";
import MoonSVG from "~/components/Icons/Moon";

export default function DarkModeIcon() {
  const darkMode = useDarkMode(false);
  const loaded = useLoaded();

  const svgSpring = useSpring({
    transform: darkMode.value ? "rotate(0deg)" : "rotate(90deg)",
  });

  return (
    <IconContainer>
      <AnimatedSpan
        style={svgSpring}
        aria-label="dark mode"
        role="button"
        tabIndex={0}
        onClick={darkMode.toggle}
        onKeyPress={darkMode.toggle}
      >
        {darkMode.value && loaded ? <MoonSVG /> : <SunSVG />}
      </AnimatedSpan>
    </IconContainer>
  );
}
