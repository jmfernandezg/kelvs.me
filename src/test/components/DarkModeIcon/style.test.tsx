import React from "react";
import { render } from "~/test/utils/index";
import "jest-styled-components";

import { IconContainer, AnimatedSpan } from "~/components/DarkModeIcon/style";

describe("Dark mode icon styles", () => {
  it("renders IconContainer correctly", () => {
    const { container } = render(<IconContainer />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders AnimatedSpan correctly", () => {
    const { container } = render(<AnimatedSpan />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
