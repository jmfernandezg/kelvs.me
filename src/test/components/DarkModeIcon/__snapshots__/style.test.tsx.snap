// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark mode icon styles renders AnimatedSpan correctly 1`] = `
.c0 {
  display: inherit;
}

.c0 svg {
  width: auto;
  height: 1.5em;
  cursor: pointer;
  fill: #232946;
}

.dark-mode .c0 svg {
  fill: #fffffe;
}

<span
  class="c0"
/>
`;

exports[`Dark mode icon styles renders IconContainer correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-self: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 93.563px;
}

@media (max-width:768px) {
  .c0 {
    display: none;
    min-width: unset;
    -webkit-flex: 0 1 20%;
    -ms-flex: 0 1 20%;
    flex: 0 1 20%;
    margin-left: 35px;
  }
}

<div
  class="c0"
/>
`;
